whirlpool/appliance.py

```diff
@@ -139,7 +145,9 @@ def get_attribute(self, attribute):
         return self._data_dict["attributes"][attribute]["value"]
 
     def has_attribute(self, attribute):
-        return attribute in self._data_dict.get("attributes")
+        if self._data_dict is not None:
+            return attribute in self._data_dict.get("attributes")
+        return None
```

Comment:
```suggestion
        if self._data_dict is None:
            LOGGER.error("No data available")
            return None
        return attribute in self._data_dict.get("attributes")
```

Reply:
2076 Traceback (most recent call last):
2077   File "/usr/src/homeassistant/homeassistant/config_entries.py", line 539, in async_unload
2078     result = await component.async_unload_entry(hass, self)
2079   File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 137, in async_unload_entry
2080     return await component.async_unload_entry(entry)
2081   File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 186, in async_unload_entry
2082     await platform.async_reset()
2083   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 745, in async_reset
2084     await asyncio.gather(*tasks)
2085   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 845, in async_remove
2086     await self.async_will_remove_from_hass()
2087   File "/usr/src/homeassistant/homeassistant/components/whirlpool/sensor.py", line 270, in async_will_remove_from_hass
2088     await self._wd.disconnect()
2089   File "/usr/local/lib/python3.10/site-packages/whirlpool/appliance.py", line 157, in disconnect
2090     await self.stop_event_listener()
2091   File "/usr/local/lib/python3.10/site-packages/whirlpool/appliance.py", line 175, in stop_event_listener
2092     await self._event_socket.stop()
2093   File "/usr/local/lib/python3.10/site-packages/whirlpool/eventsocket.py", line 166, in stop
2094     await self._run_future
2095   File "/usr/local/lib/python3.10/site-packages/whirlpool/eventsocket.py", line 125, in _run
2096     self._msg_listener("{" + match[0] + "}")
2097   File "/usr/local/lib/python3.10/site-packages/whirlpool/appliance.py", line 52, in _event_socket_handler
2098     if not self.has_attribute(attr):
2099   File "/usr/local/lib/python3.10/site-packages/whirlpool/appliance.py", line 142, in has_attribute
2100     return attribute in self._data_dict.get("attributes")
2101 TypeError: argument of type 'NoneType' is not iterable

Reply:
Ah, so the attributes key is missing.
I'll add a line suggestion to this PR to fix that.